### PR TITLE
Replace NAT with NPT

### PR DESCRIPTION
**Pull request by smorgrav**

### CODE REVIEW

**Comment by toregge** [comment]
Shouldn't this be NAPT ?

**Comment by smorgrav** [comment]
Discussed offline... I learned something about NAPT, but this is NPT (Network Prefix Translation). So much for TLAs :)

**Review by hakonhall — APPROVED** [review]
Just nits

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/docker/DockerOperationsImpl.java [hunk 1]
```diff
@@ -164,6 +168,10 @@ public void startContainer(ContainerName containerName, final ContainerNodeSpec
         }
     }
 
+    private InetAddress toPrivateSubnet(InetAddress nodeInetAddress) {
```
[the remark this answers is not in the document]
Comment (smorgrav) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/docker/DockerOperationsImpl.java]: ah..leftovers, fixed

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/docker/NetworkPrefixTranslator.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import java.net.UnknownHostException;
+
+class NetworkPrefixTranslator {
+
+    /**
+     * For NPTed networks we want to find the private address from a public.
+     *
+     * @param address    The original address to translate
+     * @param prefix     The prefix address
+     * @param subnetSize nof bits - e.g /64 subnet is 64
```
[the remark this answers is not in the document]
Comment (smorgrav) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/docker/NetworkPrefixTranslator.java]: number of - I thought once that this abbreviation was common, but clearly not. I've changed to 'in bits - e.g a /64 subnet equals 64 bits'

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/docker/DockerOperationsImpl.java [hunk 3]
```diff
@@ -120,11 +120,18 @@ public void startContainer(ContainerName containerName, final ContainerNodeSpec
...
+            if (!docker.networkNPTed()) {
+                logger.info("Network is macvlan - setting up container with public ip address on a macvlan");
                 command.withIpAddress(nodeInetAddress);
                 command.withNetworkMode(DockerImpl.DOCKER_CUSTOM_MACVLAN_NETWORK_NAME);
                 command.withVolume("/etc/hosts", "/etc/hosts"); // TODO This is probably not nessesary - review later
+            } else {
+                logger.info("Network is NPTed - setting up container with private ip address");
+                command.withIpAddress(NetworkPrefixTranslator.translate(
+                        nodeInetAddress,
+                        InetAddress.getByName("fd00::"),
```
[the remark this answers is not in the document]
Comment (smorgrav) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/docker/DockerOperationsImpl.java]: moved out the strings static

**Pull request merged**